### ISSUE TITLE
Xorg server does not shutdown

### ISSUE THREAD

**Issue by shizonic** [Earnestly/sx#1]
If I leave my window manager the Xorg still runs and does not shutdown. I have to switch to tty2 and kill it. How can I solve this?

**Comment by Earnestly** [comment]
Which window manager are you using?  I can't replicate this using simple clients such as xterm and termite (libvte) or window managers like i3, spectrwm, or ratpoison.

**Comment by shizonic** [comment]
I am using wmfs (https://github.com/xorg62/wmfs) and I use the recommended quit function ...

**Comment by Earnestly** [comment]
It seems to work as expected here when using `sx ./wmfs -C wmfsrc` and quitting with `C-M-q`.

Are you perhaps using an `sxrc`?  If so, the trouble might be there.

**Comment by shizonic** [comment]
Thanks for your reply !

Yes I Am using a sxrc file.

So you think I should report a bug to him ?

**Comment by Earnestly** [comment]
Yes, this does seem to be an issue with wmfs itself.  However you should still be returned to the tty when wmfs crashes, at least that is what happens for me on both intel modesetting and nvidia based systems.

**Comment by shizonic** [comment]
Thank you! Just created an issue for him:
https://github.com/xorg62/wmfs/issues/99

**Comment by Earnestly** [comment]
Would you mind posting your `sxrc` regardless?

**Comment by Earnestly** [comment]
xinit goes to greater lengths to terminate the server if it is taking awhile to end but it's not the kind of behaviour I want sx to have.

The commit you mention does solve the problem of segfaulting which is nice.

I'm not sure if it'll help you case.  It's hard to follow due to all the redirections and subshell usage, but I'd suggest testing a simple case first such as `sx wmfs` and build it up to identify the problem.

**Comment by Earnestly** [comment]
Ah, good to hear.  I don't have to change anything \o/

**Comment by shizonic** [comment]
No you don't. :) Is it possible that your solution is much more performant than xinit? It feels snappier...

**Comment by shizonic** [comment]
Okay. So maybe I just did imagine the better performance ;)
